2008-07-20  Adrien DELLE CAVE <adellecave@proformatique.com>

	pf_agi_astconfigpaths.patch
	$Revision$
	$Date$
	Feature

	XIVO needs Asterisk's configuration paths in AGI scripts.

Index: asterisk-1.4.21/res/res_agi.c
===================================================================
--- asterisk-1.4.21.orig/res/res_agi.c	2008-05-14 23:38:09.000000000 +0200
+++ asterisk-1.4.21/res/res_agi.c	2008-07-20 18:21:07.000000000 +0200
@@ -399,6 +399,19 @@
 	/* User information */
 	fdprintf(fd, "agi_accountcode: %s\n", chan->accountcode ? chan->accountcode : "");
     
+	/* Configuration paths */
+	fdprintf(fd, "ast_config_dir: %s\n", ast_config_AST_CONFIG_DIR);
+	fdprintf(fd, "ast_config_file: %s\n", ast_config_AST_CONFIG_FILE);
+	fdprintf(fd, "ast_module_dir: %s\n", ast_config_AST_MODULE_DIR);
+	fdprintf(fd, "ast_spool_dir: %s\n", ast_config_AST_SPOOL_DIR);
+	fdprintf(fd, "ast_monitor_dir: %s\n", ast_config_AST_MONITOR_DIR);
+	fdprintf(fd, "ast_var_dir: %s\n", ast_config_AST_VAR_DIR);
+	fdprintf(fd, "ast_data_dir: %s\n", ast_config_AST_DATA_DIR);
+	fdprintf(fd, "ast_log_dir: %s\n", ast_config_AST_LOG_DIR);
+	fdprintf(fd, "ast_agi_dir: %s\n", ast_config_AST_AGI_DIR);
+	fdprintf(fd, "ast_key_dir: %s\n", ast_config_AST_KEY_DIR);
+	fdprintf(fd, "ast_run_dir: %s\n", ast_config_AST_RUN_DIR);
+
 	/* End with empty return */
 	fdprintf(fd, "\n");
 }
